src/filepath.c

```diff
@@ -1469,6 +1469,32 @@ readdirex_checkitem(void *context, void *item)
     return retval;
 }
 
+    static int
+readdirex_dict_arg(typval_T *tv, int *cmp)
+{
+    char_u     *compare;
+
+    if (tv->v_type != VAR_DICT)
+    {
+	emsg(_(e_dictreq));
+	return FAIL;
+    }
+
+    if (dict_find(tv->vval.v_dict, (char_u *)"sort", -1) != NULL)
+	compare  = dict_get_string(tv->vval.v_dict, (char_u *)"sort", FALSE);
```

Comment:
If the key "sort" is not found, the result is unpredictable. 
```suggestion
	compare = dict_get_string(tv->vval.v_dict, (char_u *)"sort", FALSE);
    else
    {
	*cmp = READDIR_SORT_BYTE;
	return OK;
    }
```
Or, should be an error?

Reply:
Yes, I change it.

Reply:
and make missing 'sort' key an error

Reply:
Currently, the optional dict only has the 'sort' key, making it mandatory may make sense.
But if another keys are added later, making it mandatory may not make sense.